Sangeetparashar/yld 28 stakebtc revamp

sangeetparashar opened the pull request.

Sangybangy committed:
add dashboard endpoint and revamp the navbar

Sangybangy committed:
add new paths and UI for restake screen

Sangybangy committed:
fix nav bar with new menu item and new headers

Sangybangy committed:
update table for STAKE page

Sangybangy committed:
Asset addition and formatting table

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -0,0 +1,136 @@
+  };
+
+  return (
+    <div className="flex pl-[64px] pr-[56px] h-full px-6 py-0 flex-col items-start gap-4 shrink-0 self-stretch">
```
where possible, please use the tailwinds size scale instead of hardcoding. we should be leveraging their theme config to set base styling, sizing, colors, etc.

sangeetparashar replied:
removed these paddings

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -0,0 +1,136 @@
+      </div>
+
+      <div className="flex flex-col gap-2 w-full overflow-x-auto">
+        <Table.Root className="min-w-full">
```
might make more sense to create a table using radix primitives

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -0,0 +1,136 @@
+                  {asset.remainingBalance}
+                </Table.Cell>
+                <Table.Cell className="flex w-1/4 p-2 flex gap-2">
+                  <button className="bg-[#A1FD59] text-black  ">
```
colors should be in the tw theme config

text-black should be raisin black

sangeetparashar replied:
removed this logic

MattNguyen commented on src/app/stake/page.tsx:
```diff
@@ -0,0 +1,136 @@
+                      height={32}
+                    />
+                  </button>
+                  <button className="bg-[#6D655D] text-white  ">
```
same here. the text should not be #fff. It's isabelline. Nearly all of the colors are defined in the brand guide

sangeetparashar replied:
not referencing this anymore. using tailwind config file

desamtralized committed:
feat: new desktop layout for finality providers list

desamtralized committed:
wip: staking dialog

desamtralized committed:
wip: showing staking dialog with new layout and proper data

Sangybangy committed:
new staking page

desamtralized committed:
Refactor staking components and improve form handling

- Update reset prop type from boolean to number
- Implement debounced input validation for StakingTime
- Remove PreviewModal and integrate signing process into main dialog
- Adjust input types and error handling
- Improve form reset and dialog closure behavior

desamtralized committed:
better handling of stake btn while 'loading'

Sangybangy committed:
format /stake/[asset] page to fetch details about wallet and block information

Sangybangy committed:
map and generate commands for delegations in the /stake/[asset] page

Sangybangy committed:
lint fixes

desamtralized committed:
refined staking modal layout

desamtralized committed:
reponsive staking dialog

desamtralized committed:
created general dialog with new layout

desamtralized committed:
general error handling through standard dialog

desamtralized committed:
moving repeated css values to tailwind config

desamtralized committed:
fix: build errors

desamtralized committed:
using default tailwind values in simpledialog

Sangybangy committed:
smaller consumable parts of the details/delegations page

Sangybangy committed:
lint fix

Sangybangy committed:
new page for details

Sangybangy committed:
rebase and align the paths for complete flow

Sangybangy committed:
back buttons

Sangybangy committed:
lint

Sangybangy committed:
add state to refresh UI and update styling

Sangybangy committed:
logo issue on stake details page

Sangybangy committed:
update UI designs and align themes

Sangybangy committed:
dynamic actions on delegations table + style changes

Sangybangy committed:
styling and build fixes

Sangybangy committed:
Merge branch 'main' into sangeetparashar/yld-28-stakebtc-revamp

Sangybangy committed:
styling

Sangybangy committed:
linting

desamtralized commented on public/fonts/GT-America-Mono-Regular-Trial.otf:
Please remove the added otf fonts and use the existing woff fonts.

sangeetparashar replied:
removed

desamtralized commented on public/fonts/GT-America-Standard-Light-Trial.otf:
Please remove the added otf fonts and use the existing woff fonts.

sangeetparashar replied:
removed

desamtralized commented on public/eth.xsd:
Please remove unneeded xsd files.

sangeetparashar replied:
removed

desamtralized commented on public/btc.svg:
This icon is already available at `public/icons/wbtc.svg`, please remove it.

sangeetparashar replied:
removed the file

desamtralized commented on src/app/globals.css:
```diff
@@ -52,6 +52,13 @@
   font-weight: bold;
 }
 
+@font-face {
```
Please remove this declaration as it's already defined using a woff font on line 34

sangeetparashar replied:
removed

desamtralized commented on src/app/config/StakedAssets.tsx:
Please add a comment marking this file as mock data and a todo to remove it when staking balances are being properly loaded.

sangeetparashar replied:
added a todo and marked as mock data. We should use .env file or API network to fetch assets and their balances

desamtralized reviewed (changes requested):
Looks good in general, a few comments added.

Sangybangy committed:
pr comments

Sangybangy committed:
stake button fixes

Sangybangy committed:
PR comments

Sangybangy committed:
build fixes
